packages/charts/src/chart_types/metric/state/selectors/data.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ChartType } from '../../..';
+import { SpecType } from '../../../../specs';
+import { GlobalChartState } from '../../../../state/chart_state';
+import { createCustomCachedSelector } from '../../../../state/create_selector';
+import { getSpecsFromStore } from '../../../../state/utils';
+import { MetricSpec } from '../../specs';
+
+/** @internal */
+export const getSpecs = createCustomCachedSelector(
+  [(global: GlobalChartState) => global.specs],
+  (specs): MetricSpec[] => {
+    return getSpecsFromStore<MetricSpec>(specs, ChartType.Metric, SpecType.Series);
+  },
+);
```

Comment:
It may be worth combining all the simple selectors into a file to be shared throughout. I see a lot of..


```ts
const getParentDimension = (state: GlobalChartState) => state.parentDimensions;
// and
export const getSpecs = (state: GlobalChartState) => state.specs;
```

Also better to use `getMetricSpecs` to avoid name collision.

```suggestion
/** @internal */
export const getMetricSpecs = createCustomCachedSelector(
  [getSpecs],
  (specs): MetricSpec[] => {
    return getSpecsFromStore<MetricSpec>(specs, ChartType.Metric, SpecType.Series);
  },
);
```

Or even better yet create a helper function for this commonly used selector...

```ts
export const getSpecsByType<S> = (chartType: ChartType, specType: SpecType) => createCustomCachedSelector(
  [getSpecs],
  (specs): <T>[] => {
    return getSpecsFromStore<T>(specs, ChartType.Metric, SpecType.Series);
  },
);

// then just create a new selector for all
export const getMetricSpecs = getSpecsByType<MetricSpec>(
  ChartType.Metric,
  SpecType.Series
);
```

Reply:
thanks, you are right. I've done that in 96e676d276c33badfe62623a769b80c45a01bba4 and we can update the rest of the library in a different PR